src/Nethermind/Nethermind.Core/Collections/ArrayPoolList.cs

```diff
@@ -335,13 +358,18 @@ public void Dispose()
         if (!_disposed)
         {
             _disposed = true;
-            Count = 0;
             T[]? array = _array;
+            _array = null!;
             if (array is not null)
             {
-                _arrayPool.Return(_array);
-                _array = null!;
+                if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
+                {
+                    // Release any references to the objects in the array so can be GC'd.
+                    Array.Clear(array, 0, Count);
+                }
```

Comment:
This code is repeated 3x, create helper method that takes array as parameter.

Reply:
Moved to method